% ************************** Thesis Abstract *****************************

\begin{abstract}

Single-cell profiling techniques have provided an unprecented opportunity to study cellular heterogeneity at multiple molecular levels. This represents a remarkable advance over traditional bulk sequencing methods, particularly to study lineage diversification and cell fate commitment events in heterogeneous biological processes.\\
While the large majority of single-cell studies are focused on capturing RNA expression information, transcriptomic readouts provide a single dimension of cellular heterogeneity and hence contain limited information to characterise the molecular determinants of phenotypic variation. More recently, technological advances enabled multiple biological layers to be probed in parallel at single-cell resolution, unveling a powerful approach for investigating multiple dimensions of cellular heterogeneity.\\

The increasing availability of multi-modal data sets needs to be accompanied by the development of novel integrative strategies to interrogate the data generated. In this thesis I worked in collaboration with different research groups to introduce innovative experimental and computational strategies for the integrative study of multi-omics at single-cell resolution.

The first contribution of this thesis is the development of Nucleosome, Methylome and Transcriptome sequencing (scNMT-seq), a protocol for profiling RNA 
expression, DNA methylation and chromatin accessibility in single cells. This technology enables the simultaneous quantification of mRNA expression, DNA methylation and chromatin accessibility, thus providing a powerful approach for investigating regulatory relationships between the epigenome and the transcriptome at single-cell resolution. 

The second contribution is Multi-Omics Factor Analysis (MOFA), a statistical framework for the unsupervised integration of multi-omics data sets. MOFA is a Bayesian latent variable model that can be viewed as a statistically rigorous generalization of principal component analysis to multi-omics data. This statistical framework provides a principled approach to retrieve, in an unsupervised manner, the underlying sources of sample heterogeneity while disentangling axes of heterogeneity that are shared across multiple modalities from those that are specific to individual data modalities. As a proof of concept, we analysed a large (bulk) multi-omics cohort of chronic lymphocytic leukaemia patients and demonstrated how MOFA can be used to capture multiple dimensions of disease heterogeneity, enhance data interpretation and build predictive models for clinical outcomes.

The third contribution is the generation of a multi-omics roadmap of mouse gastrulation at single-cell resolution. We employed scNMT-seq to simultaneously profile mRNA expression, DNA methylation and chromatin accessibility for hundreds of cells, spanning multiple time points from the exit from pluripotency to primary germ layer specification. Using MOFA I performed an integrative analysis of the multi-modal measurements, revealing novel insights into the role of the epigenome in regulating this key developmental process. Notably, we show that regulatory elements associated with the formation of the three germ layers are either epigenetically primed or epigenetically remodelled prior to overt cell fate decisions, providing the molecular logic for a hierarchical emergence of the primary germ layers.

The fourth contribution is an extended formulation of the MOFA model (MOFA+) tailored to the analysis of large-scale single-cell data. I extended the model to incorporate a flexible regularisation that enables the joint analysis of multiple omics as well as multiple sample groups (batches and/or experimental conditions). In addition, I implemented a GPU-accelerated stochastic variational inference framework, thus enabling the scalable analysis of potentially millions of samples.

\end{abstract}
