% ************************** Thesis Abstract *****************************

\begin{abstract}

Single-cell profiling techniques have provided an unprecented opportunity to study cellular heterogeneity at multiple molecular levels. This represents a remarkable advance over traditional bulk sequencing methods, particularly to study lineage diversification and cell fate commitment events in heterogeneous biological processes.\\
While the large majority of single-cell studies are focused on capturing RNA expression information, transcriptomic readouts provide a single dimension of cellular heterogeneity and hence contain limited information to characterise the molecular determinants of phenotypic variation. More recently, technological advances enabled multiple biological layers to be probed in parallel at single-cell resolution, unveling a powerful approach for investigating regulatory relationships. Such single-cell multi-modal technologies can reveal multiple dimensions of cellular heterogeneity and uncover how this variation is coupled between the different molecular layers, hence enabling a more profound mechanistic insight than can be inferred by analysing a single data modality.\\
The increasing availability of multi-modal data sets needs to be accompanied by the development of novel integrative strategies. Yet, the high levels of missing information, the inherent amounts of technical noise and the potentially large number of cells, makes the integrative analysis of single-cell omics one of the most challenging problems in computational biology.

The main contribution of this PhD thesis is the introduction of computational strategies for the integrative study of multi-modal measurements at single-cell resolution.

% DESCRIBE MORE IN DETAIL

\end{abstract}
