\graphicspath{{Chapter3/Figs/}}

\chapter{Single cell multi-omics profiling reveals a hierarchical epigenetic landscape during mammalian germ layer specification}

\section{Introduction}

The human body is composed of a myriad of cell types with specialised structure, organisation and function; and yet, each cell in the body contains the same genetic information. The modulation of the genetic code by internal and external factors begin during embryonic development, giving rise to the formation of specialised molecular patterns that ultimately determines the complexity of adult organisms \cite{Rosalind2018}.
A key phase in mammalian embryonic development is gastrulation, when a single-layered blastula of pluripotent and relatively homogeneous cells is reorganised to form the three primordial germ layers: ectoderm, mesoderm and endoderm \cite{Tam1997, Solnica-Krezel2012, Tam2007}.



\subsection{Gastrulation overview}
The onset of gastrulation is determined by the formation of the primitive streak, which establishes the initial bilateral symmetry of the body. Involution of epiblast cells through the primitive streak gives rise to the mesoderm and endoderm, whereas epiblast cells establish the ectoderm. In the mouse embryo, gastrulation begins around embryonic day E6.5, which establishes the antero-posterior axis by the migration of primitive streak cells to the posterior region of the embryo \cite{Arnold2009,Tam2007,Tam1997,Tam1993}. Although  differences exist between species, the morphogenic process of gastrulation is evolutionary conserved throughout the animal kingdom \cite{Solnica-Krezel2012}.\n
In most cases, gastrulation is characterised by an epithelial to mesenchymal transition that brings mesodermal end endoderm progenitors beneath the future ectoderm.The epiblast cells that did not migrate through the primitive streak differentiate towards ectoderm, which eventually gives rise to the nervous system (neural ectoderm) and epidermis (surface ectoderm). The embryonic endoderm gives rise to the interior linings of the digestive tract, the respiratory tract, the urinary bladder and part of the auditory system. The embryonic mesoderm gives rise to muscles, connective tissues, bone, cartilage, blood, kidneys, among others \cite{XX}. 
The cell fate map of embryonic cells has been an area of intensive research \cite{XX}. Early work in model organisms such as \textit{Drosophila melanogaster} identified morphogenic gradients that partition the embryo into multiple compartments, and specific combinations of extracellular molecules will induce differential signalling cascades that will commit cells to the different cell type fates. A classic example are Bone Morphogenic Proteins (BMPs) and their antagonist Fibroblast Growth Factors (FGFs), which help establish a patterning in the embryo that will induce different mesoderm fates. Dorsal BMP-rich regions will commit towards blood and vasculature, whereas ventral BMP-low regions will commit towards somites \cite{Dorey2010}.

More recently, advances in CRIPSR-Cas9 genome editing have revolutionised lineage tracing methods. By introducing genetic scars with high-information content, this technology promises a big leap towards the ultimative goal of creating a high-resolution fate map for every cell in the embryo \cite{McKenna2019,Chan2019}.


\subsection{Transcriptomic studies}
Significant research effort has been deployed to understand the molecular changes underlying gastrulation. Historically, microscopy was used to quantify gene expression at single cell resolution \cite{XX}. However, constraints imposed by fluorophore emission spectra made this approach unsuitable for genome-wide studies. Only after the breakthrough of single-cell sequencing technologies it has been possible to generate comprehensive molecular roadmaps of differentiation processes \cite{Schaum2018,PijuanSala2019,Cao2019,Regev2018}.

In a pioneer study, \cite{PijuanSala2019} generated the first high-resolution atlas of gastrulation and early somitogenesis by profiling the RNA expression of 116,312 cells from 411 whole mouse embryos collected between E6.5 and E8.5. This effort completed earlier attempts of reconstructing the transcriptomic landscape of post-implantation embryos \cite{Mohammed2017,Scialdone2016,Ibarra-Soria2018,Wen2017}. At the same time, another study employed single-cell combinatorial indexing to profile around 2 million cells from 61 embryos ranging from E9.5 and 13.5 days of gestation, spanning early organogenesis \cite{Cao2019}. By constructing a densely sampled reference data set, both works have let the ground for studying the molecular consequences of developmental mutants.
Computational approaches for matching heterogeneous data sets \cite{Haghverdi2018,Stuart2019} can be employed to rigurously annotate cell types in new data sets, hence providing a framework to quantify the effect of a wide range of experimental perturbations, both \texit{in vitro} and \textit{in vivo}.



\subsection{Beyond the transcriptome: epigenetic regulation}

With the publication of the mouse gastrulation atlas the gene expression dynamics of gastrulation have been largely resolved. However, the information contained in RNA expression data is only a piece in the puzzle, and the next step is connecting it to the accompanying epigenetic changes. In differentiated cell types, epigenetic marks confer stable characteristic patterns of cell type identity which have been extensively profiled using bulk sequencing approaches. However, because of the low amounts of input material and the extensive cellular heterogeneity, the study of the epigenome landscape during early development remains poorly understood \cite{Kelsey2017}.

\subsubsection{Pre-implantation: establishment of the pluripotent state}
The first efforts to interrogate the epigenetic dynamics using (bulk) next generation sequencing technologies have provided valuable insights for the pre-implantation stage. Multiple studies have described that, after fertilisation, there is a round of reprogramming that resets the epigenetic landscape to a ground state \cite{Smith2012,XXX}. DNA methylation is globally removed and the chromatin attains its highest levels of accessibility \cite{Wu2016}. Consistently, Hi-C experiments reported a very flexible chromatin landscape, with lack of topologically associating domains (TADs) and chromatin compartments \cite{Ke2017,Du2017,Tee2014}. All together, this \texit{plain} epigenetic landscape is a very likely explanation for the plasticity of pluripotent ESCs.\\
Interestingly, in Chapter 1 of this thesis we show that in naive ESCs is when the correlation of both epigenetic layers and RNA expression is at its lowest. This raises the question of whether DNA methylation and chromatin closure play a role in the regulation of RNA expression at this stage. Consistently, knock-out experiments of Dnmts in ESCs display no obvious phenotype, but it does prevent exit from pluripotency and differentiation \cite{XXX}.

In contrast to DNA methylation, the presence of post-translational modifications in histone marks are abundant at this stage, potentially providing the major mechanism of epigenetic regulation \cite{Hanna2018,Tee2014}. Several histone modifications have been studied in ESCs, the most prominent being H3K27ac and H3K4me3, both (generally) activatory marks; and H3K27me3 and H3K9me3, both (generally) repressive marks \cite{Zhao2015}. Interestingly, many genes that are silenced in ESCs contain both activatory (H3K4me3) and repressive (H3K27me3) epigenetic marks. This distinctive signature of ESCs is though to establish a bivalent or poised signature for a transcriptionally-ready state for genes that become expressed after gastrulation \cite{Bernstein2006,Tee2014}. 


\subsubsection{Post-implantation: exit of pluripotency}
In post-implantation developments, cells exit pluripotency and (arguably) undergo the first set of embryonic cell fate decisions that will ultimately give rise to the myriad of somatic cell types. Yet, while multiple studies have interrogated the epigenetic landscape in pre-implantation embryos, the epigenetic landscape of gastrulation and early mammalian organogenesis remains largely unexplored.\\

DNA methylation is one of the few epigenetic marks that has been profiled in a genome-wide approach in post-implantation mouse embryos, both in bulk and at the single cell level \cite{Auclar2014,Zhang2017,Dai2016,Rulands2018}.
All studies found that the hypomethylated state in E3.5 blastocysts is followed by a \textit{de novo} DNA methylation wave upon implantation (between E4.5 and E5.5) that leads to a hypermethylation of most of the genome, although long stretches of CpG-rich sequences are partially protected from the DNA methylation wave (CpG islands). These dynamics are well-replicated in ESCs when switched from 2i media to serum, hence providing an excellent \textit{in vitro} model.\\
The increase in DNA methylation is concomitant with the increased deposition of repressive histone marks, presumably with the aim of restricting the differentiation potential of early pluripotent cells \cite{Atlasi2017}.

As described in Chapter 1, the \textit{de novo} methyltransferases (DNMT3A and DNMT3B) are the enzymes responsible for the insertion of DNA methylation marks. Both genes are highly expressed in ESCs, and catalytically inactive mutants of both enzymes lacked \textit{de novo} methylation activity \cite{Auclair2014,Okano1999}. Interestingly, mouse ESCs remain viable despite complete loss of DNA methylation, but they are uncapable of escaping from the pluripotent state \cite{XXX}. Inactivation of one of the enzymes shows only a partial reduction in global DNA methylation, indicating that they share substantial redundancy. However, the effect of the Dnmt3b KO is more severe and affects imprinted genes that are critical for the deregulation of the germ line, resulting in drastic developmental defects.\\

The interplay of histone marks during post-implantation development is complex and remains poorly understood. H3K4me3 is detected at transcription start sites after the zygotic genome activation, and remains one of the most stable histone marks across different pluripotency stages as well as cell types \cite{XXX}. It is thought to facilitate transcription by inducing a more efficient assembly of the transcriptional machinery \cite{xxx}. The other conventional activatory mark, H3K27ac, is deposted in different types of regulatory elements, including promoters and enhancers. It is significantly more dynamic than H3K4me3 in response to internal and external stimuli, and is hence a strong candidate to regulate cell fate transitions \cite{XXX}.\\
	(TO-FINISH)H3K27me3, the conventional inhibitory mark, shows a marked increase upon post-implantation.
	(COPIED. TO FINISH) In mouse embryos, H3K27me3 is gradually gained from the 2-cell stage to the 16-cell stage, followed by a particularly marked increase during the transition from the morula to the early blastocyst stage27. In serum-cul- tured mouse ESCs, H3K27me3 decorates promoters of developmental genes59,60. Depleting components of Polycomb repressive complex 2 (PRC2; the writer of the H3K27me3 mark) in these cells leads to activation of PRC2 target genes and increased spontaneous cell differ- entiation61,62. By contrast, H3K27me3 is strongly reduced at these promoters in 2i-cultured ESCs63 (BOX 1; FIG. 1), but surprisingly without causing transcriptional activa- tion12,64. This observation suggests that either unknown mechanisms can substitute for H3K27me3 function in gene silencing or that 2i-cultured cells do not express the transcription factors required for transcriptional activa- tion of these promoters. PRC2 is dispensable for mouse ESC self-renewal65–67, and depleting PRC2 components does not affect pre-implantation development, although embryos lacking PRC2 die shortly after gastrulation68–70
	(COPIED) Taken together, these observations suggest that H3K27me3 is dispensable for transcriptional repres- sion in naive mouse ESCs12,75,76 but is essential at later developmental stages, particularly in post-implantation mouse embryos. Accumulating data indicate that dep- osition of H3K27me3 marks might occur after gene silencing, as transcriptional repression results in the recruitment of the PRC2 complex75. Moreover, mouse ESCs lacking PRC2 show dysregulated transcriptional repression only after long-term differentiation, which suggests that H3K27me3 is required for maintaining, rather than initiating, gene silencing75.


\subsubsection{Gastrulation: germ layer specification}
The post-implantation blastocyst is relatively homogeneous and can be accurately characterised by bulk sequencing approaches. However, germ layer specification is uniquely heterogeneous and very difficult to study without single-cell technologies.

MENTION SINGLE-CELL EPIGENETIC TEHCNLOGIES ARE DESCRIBED IN CHAPTER 1

EXPLAIN DEMETHYLATIONE VENTS FROM ZEBRAFISH: TET PROTEINS\tabularnewline
(COPIEDE) DNA methylation is a reversible process. DNA demethyl- ation is achieved either by passive replication-dependent dilution of methylation or by active enzymatic demeth- ylation100. In the case of active enzymatic demethylation, 5-methylcytosine (5mC) undergoes a stepwise chemical oxidation by the methylcytosine dioxygenase TET pro- teins (TET1, TET2 and TET3) that ultimately leads to removal of the mark90. 5-Hydroxymethylcytosine (5hmC) is the best-studied and first intermediate of this enzymatic cytosine demethylation. 5hmC is mainly found at enhanc- ers, and depleting the three TET enzymes in mouse ESCs results in increased DNA methylation at 15–25% of all enhancers101. Tet1 and Tet2 are highly expressed in mouse ESCs and are downregulated on cell differentiation102, whereas Tet3 mainly demethylates the sperm genome after fertilization84. Mice lacking both TET1 and TET2 are viable and overtly healthy103. Mouse ESCs that lack all three TET proteins maintain self-renewal, despite their complete lack of 5hmC; however, these cells show differentiation defects and fail to contribute to chimeric embryos, which indicates a role for TET proteins in ESC differentiation104. Thus, both DNA methylation and TET function are dispensable for mouse ESC self-renewal but are required for exit from pluripotency and for proper differentiation11,13,88,103–105. The rapid post-fertilization demethylation of the sperm genome was initially explained by TET3-driven oxidation of 5mC to 5hmC84,85. However, more recent findings show that depleting Tet3 from mouse embryos prevents accu- mulation of 5hmC but does not influence early demeth- ylation of the paternal genome106. The existence of an alternative unknown mechanism for DNA demethylation in the early mouse zygote has therefore been suggested106, but the exact role of Tet3 and 5hmC in this process remains to be addressed. In contrast to the paternal genome, mater- nal DNA has been shown to lose methylation mainly via replication-dependent demethylation in mice100

Recent studies have identified germ layer specific chromatin marks, accessibility and DNA methylation profiles at regulatory elements in several species26–31. Together, this suggests that correct establishment of chromatin accessibility and DNA methylation may be important for cell fate specification during development. Indeed, mutants that fail to correctly remodel their epigenetic landscape display differentiation defects at or following gastrulation32,33. However, since cell fate decisions are made at the level of single cells, the ability to understand whether distinct epigenetic environments precede or follow cell fate choice during early development has been challenging. Moreover, in isolation, characterising the epigenome of individual cells provides only part of the picture – without knowledge of the transcriptional readout, associating epigenetic changes with specific cell fate choices is difficult13.


	

	(cOPIED) During differentiation, acquisition of H3K27ac at enhancers is associated with the formation of enhancer-promoter interactions and induction of their target genes (Wang et al., 2016; Rubin et al., 2017). Knockout in mice of the H3K27 acetyltransferases CBP or p300 (which also acetylate other histone residues and interact with many transcription factors themselves) leads to mid-gestation embryonic lethality (Yao et al., 1998). Surprisingly, this suggests that H3K27ac is not required for establishing the transcriptional programming during early lineage specification.

	(COPIED) Together these studies suggest that several epigenetic marks, in particular H3K27me3 and H3K4me3, are required for lineage specification, but for the most part it appears that epigenetic modifications may reinforce lineage commitment rather than direct i


(4) Describe recent technological advances and our study

(COPIED)




Recent technological advances have enabled the profiling of multiple molecular layers in single cells34–43, providing novel opportunities to study the relationship between the transcriptome and epigenome during cell fate decisions. Here we apply scNMT-seq (single-cell Nucleosome, Methylome and Transcriptome sequencing) to comprehensively analyse the collective dynamics of RNA expression, DNA methylation and chromatin accessibility in peri-implantation and early postimplantation mouse embryos during exit from pluripotency  and primary germ layer specification. 



(TO-ADD) Novel lineage tracing approaches combined with high-throughput sequencing promise to deliver unpredecented knowledge