\graphicspath{{Chapter5/Figs/}}

\chapter{Concluding remarks}

The last few years have experienced an explosion of single-cell sequencing technologies, with the consequent opening of new directions and opportunities to understand biological complexity. The ultime goal of single-cell sequencing is to move from the descriptive snapshots to comprehensive multi-modal roadmaps of biological processes mapped across time and space. Unifying molecular variation across these two dimensions will be at the forefront of scientific research.
 
The first stones on this direction have been laid. Experimental designs that include single-cell genomics technologies have now become ubiquitious, and the computational pipelines are gradually becoming standarised. In addition, multi-modal measurements have been successfully assayed in single cells, although this remains in pilot stages and at the time of this writing very few commercial platforms are available, thus limiting its widespread use by the community. 

In this Chapter we discuss current and future perspectives of experimental and computational methods that will lay the foundations for an unprecedented and exciting stage to study biological complexity using single-cell multi-omics.


\section{Experimental perspectives} 

\subsection{Recording space} 

The dissociation and pooling of cells from their native location, with the consequence loss of information on their spatial coordinates, is one of the biggest limitations of current single-cell technologies. The positioning and the interaction of cells in their native tissue is essential to understand biological function, as complex tissues arise following organized events in space and time \cite{Mayr2019}. Pioneering work in single-molecule fluorescence \textit{in situ} hybridization demonstrated that mRNA molecules can be measured, but quantifying multiple genes at the same time has been a major challenge, mostly hindered by physical limitations on the optical resolution and the high density of transcripts within each cell \cite{Eng2019}. Remarkbly, tecent technological breakthroughts have scaled these assays to transcriptome-wide measurements while maintaining high accuracy. One of these methods, seqFISH \cite{Lubeck2014,Eng2019}, employs a multiplexed strategy to overcome the diffraction limit where multiple rounds of sequential probe hybridization and imaging are applied. Although most of the spatially-resolved methods have been focused on transcriptomics, epigenomic measurements have also been successfully recorded by adapting the ATAC-seq protocol \cite{Thornton2019}, thus paving the way for future groundbreaking multi-modal spatially-resolved assays.

\subsection{Recording time} 

\subsubsection{Recording the past} 

In the timespan of a few days, a mammalian embryo expands from a handful of cells to milions of them, experiencing on the process a set of cell fate commitments that will eventually generate a myriad of specialised cell types. Besides knowing the spatial location of cells, recording the timing and the hierarchy of the events at high-resolution is an essential step to understand the complex biological dynamics.  As we have demonstrated in this thesis, static single-cell experiments can provide snapshots that can be used to reconstruct dynamic differentiation processes \cite{Weinreb2018}, but the past story of the cells remains elusive.

A fundamental principle of biology is that each cell originates from another existing cell. Thus, each adult cell has an associated cell lineage tree that unambiguously defines its past history and can potentially be recorded by tracking the progeny of single cells. Historically, imaging-based techniques have been used to perform lineage tracing in a low-throughput manner by employing fluorescent protein markers. However, classical fluorescence-based experiments are limited when it comes to both temporal and molecular resolution. The current generation of lineage tracing techniques introduce inheritable genetic marks that can be read in terminal cells through next-generation sequencing \cite{Baron2019,Kester2018,McKenna2019}. For example, one can induce double-stranded breaks using CRISPR-Cas9 at target genomic sites, which after repair results in random genomic insertions and deletions (indels) that become inherited in different hierarchical combinations by the progeny cells \cite{Baron2019,Kester2018,McKenna2019}. Notably, this strategy can be combined with single-cell sequencing for the simultaneous quantification of clonal history (applying phylogenetic methods on the indel profiles) and cell type identity (applying clustering methods on the gene expression profiles). This strategy was employed to map cell fates in several model organisms, including adult zebrafish \cite{Alemany2018} an mouse development \cite{Chan2019}. 

One of the most exciting opportunities that soon will become a reality is combining lineage tracing technologies with single-cell multi-modal readouts. An ideal system would measure multiple biological layers \textit{in situ} and would recover each cell's past history at the same time \cite{McKenna2019}. In the context of this thesis, the genomic modality of scNMT-seq could be employed to track not only CRISPR-Cas9-induced indels but also naturally occurring mitochondrial mutations. The latter in particular would be critical to identify clones in complex human tissues, where genetic manipulation is not an option \cite{Ludwig2019}. 

In conclusion, multi-modal lineage tracing would be a major step towards the ultimate goal of constructing dynamic models of cell fate commitment and to understand how they become dysregulated in disease. Notably, each of these tools are already available, but combining them in coassays will be one of the big challenges for the next years.

\subsubsection{Recording the future} 

scRNA-seq offers transcriptome-wide snapshots of the present status of each cell. However, simultaneously measuring some property of its future state would provide extremely valuable information to understand cell fate decisions. 

A milestone of scRNA-seq analysis was achieved with the inference of RNA velocity in single cells \cite{LaManno2018}. This method leverages the small quantity of intronic reads that are present in the sequencing library to calculate a relative ratio of unspliced (intronic) and spliced (exonic) mRNAs to infer gene expression kinetics. In turn, by pooling information across multiple genes and under some assumptions one can estimate the nascent transcriptional state for each cell. Interestingly, similar information can be obtained from spatial transcriptomics data. However, instead of identifying intronic reads, which are lowly abundant due to polyA selection, one can exploit the subcellular resolution of the molecules to distinguish between nuclear (unspliced) and cytoplasmic (spliced) reads \cite{Xia2019}.

RNA velocity is appealing because it can be applied to most conventional scRNA-seq protocols. However, intronic coverage is sparse, particularly for droplet-based assays, thus limiting the reliability of the method in some data sets \cite{Soneson2020}. Moreover, the inference of gene expression kinetics is restricted to genes that undergo splicing events. Alternative strategies to record the future state of individual cells are aimed at directly monitoring newly synthesised RNA. For example, NASC-seq \cite{Hendriks2019} relies on chemical modifications of the nascent RNA that can be read out by sequencing at much greater sensitivity and better temporal resolution than RNA velocity estimates.

% An interesting future direction will again be to combine multi-modal assays such as SNARE-seq, SHARE-seq or scNMT-seq with nascent RNA labeling. This would be extremely useful to study epigenetic mechanisms of gene expression regulation.


\section{Computational perspectives} 

None of the biological insights offered by multi-modal assays would be possible without concomitant development of computational methods. Each new data modality presents distinct challenges, from low level processing, quality control and normalisation through to downstream challenges such as quantifying sources of biological variability and using these to generate testable biological hypotheses. Additionally, in the context of single-cell multi-omics datasets there exist specific challenges that need to be overcome: the high levels of missing information, the inherent amounts of technical noise and the potentially large number of cells. Here I discuss some directions that need to be addressed in order to extract the best out of single-cell multi-omics.

\subsection{Mechanistic insights}

One of the most promising aspects of multi-modal sequencing is the opportunity to move from descriptive snapshot to a more mechanistic understanding of gene regulation. By incorporating prior knowledge about the hierarchical relationship between molecular layers, we envision that multi-modal assays will play an important role in identifying causal chains of events in gene regulatory networks. However, to construct such mechanistic models it will be essential to combine multi-modal readouts with perturbations assays.

Single-cell perturbation studies using the CRISPR technology is one of most exciting experiments that single-cell genomics has brougth us \cite{Dixit2016,Datlinger2017,Jaitin2016,Alda-Catalinas2020}. In the first step, cells are infected with a pool of lentiviral constructs that contain guide RNAs that target (typically by inactivation) specific genes. Notably, increasing the multiplicity of infection during infection can be used to target multiple genes at once and thus study epistatic effects. After stimulation or differentiation, cells are sequenced using single-cell  methodologies. The unique combination of barcodes within each cell enables the computational identification of the gene(s) targeted. Remarkably, this strategy of pooling guides and cells together and later deconvoluting them enables this protocol to be done in a massively parallel fashion.

While most of the perturbation studies have been focused on RNA expression, the profiling of epigenomic layers is receiving increased interest \cite{Rubin2019}. Yet, to my knowledge, no multi-modal single-cell CRISPR screening has performed to date. This is a matter of time, as all the ingredients are already available, and I envision that this will become the state-of-the-art for the characterisation of gene regulatory networks.


\subsection{Benchmarking of methods} 

Benchmarking of methods has been extensively performed on \textit{horizontal} data integration strategies, particularly in the context of batch correction \cite{Luecken2020,Tran2020}. However, benchmarking \textit{vertical} and \textit{diagonal} integration strategies is notorously difficult, as ground truth is rarely known. In the context of MOFA, for example, it is difficult to assess the quality of the output. There are useful quality control metrics, such as the number of factors or the total variance explained, but it is not clear how to assess whether the latent Factors are a reliable representation of biological variation or whether they are just arbitrary (but useful) mathematical representations.

In this context, having gold-standard truth data sets will probe essential to benchmark integrative methods. Here, we discuss two existent biological systems that are well suited to benchmark data integration tasks. The first one is Peripheral Blood Mononuclear Cells (PBMCs), which is the \textit{de facto} dataset to validate single-cell technologies developed by 10x Genomics, owing to its simplicity and well-characterised subpopulations of cells. Multiple assays have been profiled on PBMCs across multiple human donors and different species, including scRNA-seq, scATAC-seq, CITE-seq and [T/B]-cell receptor sequencing. Moreover, some horizontal and vertical integration strategies have already been successful applied \cite{Stuart2019b}.

The second biological system is mammalian embryonic development, a significantly more complex system with branching differentiation trajectories and where the regulation between molecular layers is less well understood as compared to somatic cell types. In addition, unlike PBMCs, the solid tissue enables the integration of molecular readouts with spatial context information. A large variety of single-cell technologies have been applied to mouse embryonic development, including scRNA-seq \cite{Pijuan-Sala2019,Mohammed2017,Scialdone2016,Grosswendt2020,Nowotschin2019}, scATAC-seq \cite{Pijuan-Sala2020}, scNMT-seq \cite{Argelaguet2019} and even spatial measurements \cite{Peng2019,VanDenBink2020}.

As the Human Cell Atlas project \cite{Aviv2017} matures, we expect many more biological systems to be suitable data sets for data integration not only across data modalities but also across individuals and even across different species.

\subsection{Mosaic integration} 

A major challenge in the next few years will be to integrate independent experiments with the goal of building self-consistent multi-modal data sets of biological processes. However, given how difficult it is to simultaneously capture multiple molecular layers in an efficient and scalable manner, this task will require computational integration of independent uni-modal and multi-modal experiments from the same biological system. There is an urgent need to develop a unifying integrative strategy that selectively exploits cells and features as common coordinate frameworks to perform transfer learning of molecular signatures across experiments (\Cref{fig:mosaic_integration}). I coin the term \textit{mosaic} integration for this combination of \textit{vertical}, \textit{horizontal} and \textit{diagonal} integration tasks. Undeniably, this task will not be solved by linear matrix factorisation frameworks such as MOFA, and it will require non-linear strategies probably in the form of multi-view variational autoencoders \cite{Lopez2018}.

\begin{figure}[H]
	\centering
	\includegraphics[width=0.95\linewidth]{mosaic_integration}
	\caption[]{
	\textbf{Mosaic integration} The aim is to integrate a complex experimental design that consists of uni-modal and multi-modal data sets by selectively exploiting cells and features as common coordinate frameworks. The output would be a self-consistent data set where all missing data modalities have been imputed across all experimental conditions.}
	\label{fig:mosaic_integration}
\end{figure}

% \subsection{Multi-scale modelling} 

\subsection{Software infrastructure} 

Open-source software, data sharing platforms and reproducible analysis pipelines is essential in computational biology. Generic frameworks that can contain increasingly large complex experimental designs from single-cell genomics are urgently needed. Popular tools for scRNA-seq analysis such as \textit{SingleCellExperiment}, \textit{Seurat} and \textit{Scanpy} are continuously being extended to handle novel multi-modal assays. There are however important challenges that need to be overcome. The first one is data standarisation and interoperabiliy between platforms. Conversion between R-based objects is relatively easy, but connecting Python and R is still a significant challenge that involves tedious configurations \cite{reticulate}. Second, the large-scale of single-cell genomics requires optimisation of memory usage. Just to give a sense of how important this is, one of data sets released by 10x Genomics contains transcriptome-wide measurements for 1.3 million cells from the mouse brain \footnote{\url{https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.3.0/1M_neurons}}. Storing this data set in an ordinary integer matrix requires more than 100 GB of memory. Clearly, the conventional approach of loading the entire data set to memory in a persona laptop becomes prohibitive. One of the most promising approaches to handle vast amounts of data is to use on-disk operations, where common array operations are performed using a block processing mechanism, thus preventing the entire object from being loaded in memory. This is a strategy that we implemented for the downstream analysis in MOFA by adapting the \textit{DelayedArray} framework \cite{Herve2020}, albeit the matrix operations during model training require loading the entire data set in memory. Finally, we need centralised data sharing platform where curated landmark data sets are made available with a common data fromat and alongside reproducible vignettes. 

\pagebreak

\section{Thesis summmary}

In this Thesis I have described the work performed throughout my PhD where I sought to develop and apply computational strategies for data integration in the context of single-cell multi-omics.

First, I worked together with experimental collaborators to devise scNMT-seq, an experimental protocol for the genome-wide profiling of RNA expression, DNA methylation and chromatin accessibility in single cells. I developed the entire computational analysis pipeline and after validatind teh quality of the readouts I demonstrated how scNMT-seq can be used to study coordinated epigenetic and transcriptomic heterogeneity along a simple differentiation process.

Second, motivated by the need to discover biologically meaningful insights from such complex data I developed Multi-Omics Factor Analysis (MOFA), a statistical framework for the integration of multi-omics data sets. Briefly, MOFA is a statistically rigurous generalisation of Principal Component Analysis for multi-view data. It provides a systematic approach to explore, in an unsupervised manner, the underlying sources of sample heterogeneity in a multi-omics data set. Owing to its linear formulation, interpretability is an essential property of this model that permits a series of useful downstream analysis. Before applying MOFA to single-cell data sets, we benchmarked it using a large multi-omics cohort of chronic lymphocytic leukaemia patients and demonstrated how MOFA can be used to capture multiple dimensions of disease heterogeneity, enhance data interpretation and build predictive models for clinical outcomes.

Third, I aimed to leverage the experimental and computational frameworks described above to study embryonic development and specifically germ layer commitment. Together with experimental collaborators we employed scNMT-seq to simultaneously profile mRNA expression, DNA methylation and chromatin accessibility for more than 1000 cells, spanning four time points from the exit from pluripotency to primary germ layer specification. This data set represents the first multi-omics roadmap of mouse gastrulation at single-cell resolution, which enabled us to perform an integrative study that revealed novel insights on the dynamics of the epigenome. Notably, we show that cells committed to mesoderm and endoderm undergo widespread epigenetic rearrangements, driven by demethylation in enhancer marks and by concerted changes in chromatin accessibility. In contrast, the epigenetic landscape of ectoderm cells remains in a \textit{default} state, resembling earlier stage epiblast cells. This work provides a comprehensive insight into the molecular logic for a hierarchical emergence of the primary germ layers, revealing underlying molecular constituents of the Waddington's landscape.

Finally, after having benchmarked MOFA using high-quality bulk multi-omics and relatively small single-cell genomics dat asets, I developed a second version of the software aimed at the scalable analysis of datasets with thousands of samples and more complex experimental designs. Key methodological improvements included a fast stochastic variational inference framework and a flexible structure of the prior distributions that enable integration of multiple groups of samples. 
